backend/libraries/ballerina-core/business rule engine/parser/expr.fs

```diff
@@ -416,6 +416,20 @@ module Expr =
           |> state.MapError(Errors.WithPriority ErrorPriority.High)
       }
 
+    static member private ParseList<'config, 'context>(json: JsonValue) : State<Value, 'config, 'context, Errors> =
+      state {
+        let! fieldsJson = assertKindIsAndGetFields "list" json
+
+        return!
+          state {
+            let! elementsJson = fieldsJson |> sum.TryFindField "elements" |> state.OfSum
```

Comment:
no need for all these intermediate variables, can pipe the whole thing til the end

Reply:
`State.bind` is wrong, need to fix that one first

Reply:
https://github.com/Ballerina-Org/ballerina/pull/225

Reply:
Discussed, to be left as its

Reply:
:/